### ISSUE TITLE
All conv layers have zero gradients and images are black

### ISSUE THREAD

**Issue by jbmaxwell** [debtanu177/CVAE_MNIST#1]
I just gave the cvae a try and the conv layers seem to just be outputting zeros. Strange:

```
conv3.weight tensor([[0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.]], device='cuda:0') 
conv3.bias tensor(0., device='cuda:0') 
conv4.weight tensor([[0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.],
        [0., 0., 0., 0., 0.]], device='cuda:0') 
conv4.bias tensor(0., device='cuda:0') 
conv5.weight tensor([[0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.],
        [0., 0., 0., 0.]], device='cuda:0') 
```

Ran for 14 epochs and all images are black.
Maybe a PyTorch compatibility problem? I'm on 1.6.

**Comment by jbmaxwell** [comment]
Add batch norm solved it.
